### DISCUSSION TITLE
How can I disable Pyright diagnotic function?

### DISCUSSION THREAD

**Discussion by stephane-klein** [microsoft/pyright#3929] [Q&A] [answered]
Hi,

I use Neovim and I would like to disable `pyright` diagnostics to use instead [`null-ls`](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/MAIN.md) with [flake8](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md#flake8). For more context, [see this thread](https://old.reddit.com/r/neovim/comments/vz30kb/whats_your_python_editing_setup/inzs7r2/).

Question: how can I disable Pyright diagnotic function?

Best regards,  
Stéphane

**Comment by erictraut** [comment, marked as answer]
Pyright is a static type checker, whereas flake8 is a linter. They have some overlap, but they are designed for somewhat different purposes. Pyright also includes some language server features (hover text, simple completion suggestions, etc.). If I understand you correctly, you're interested in using the language server features without the type checking features.

Many of pyright's diagnostics can be disabled by specifying the severity level of "none" in the `pyrightconfig.json` file. You can also disable all type-checking diagnostics by setting `typeCheckingMode` to "off" in the configuration. There are some diagnostics (e.g. syntax errors) that cannot be disabled. For details about configuration, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/configuration.md).

**Reply by Hubro** [reply]
Any way to disable every diagnostic using only the LSP settings? I want to disable everything by default in my editor config and I don't want to have to create a config file for every project.

I only use the LSP for auto-completion, auto-imports, inline documentation and go-to-definition. For diagnostics, I use Ruff and MyPy. This basically completely overlaps with Pyright, so I'd rather disable everything from Pyright.

**Reply by erictraut** [reply]
Yes, you can set `python.analysis.typeCheckingMode` to "off" in your LSP settings.

**Reply by Hubro** [reply]
I currently have these set:

```lua
settings = {
  python = {
    analysis = { diagnosticMode = "off", typeCheckingMode = "off" },
  },
},
```

I'm still getting warnings like

> "console" is not accessed

**Reply by erictraut** [reply]
Those are not warnings. They're hints that have a special tag indicating that the text should be displayed in a grayed-out manner to indicate that it is not referenced. Pyright assumes that the editor will not display these as normal diagnostics, but some editors do. Pyright provides no way to disable these. You may need to configure your editor to ignore all "hints" if you don't want to see these. Or if your editor allows you to specify that it doesn't handle the "unreferenced_code" tags, pyright will not send these to the client.

**Reply by burnettk** [reply]
I wouldn't want to configure my editor to ignore all "hints" since hints are awesome or to instruct the editor that it shouldn't handle the "unreference_code" tags since these are quite helpful. I just don't want the same hint twice, once from null-ls (and ruff in this case) and once from pyright. would there be any chance that this could become a pyright configuration option in the future? thanks very much for the awesome tool and sorry for commenting on an old thread, but it was a top hit on my search engine.

**Comment by shinonomeow** [comment]
i meet same question,
module is no accessed appear two times(ruff and pyright).
do you solve the question?

**Comment by epsentosa** [comment]
i give up on trying to use pyright only for LSP, not the diagnostics
so my solution now to use jedi as lsp completion and ruff as diagnostics

```lua
local lspconfig = require("lspconfig")
lspconfig.jedi_language_server.setup({
  capabilities = capabilities,
  on_attach = on_attach,
  handlers = handlers,
  init_options = {
    completion = {
      disableSnippets = true,
    },
  }
})
lspconfig.ruff_lsp.setup({
  capabilities = capabilities,
  on_attach = on_attach,
  handlers = handlers,
  init_options = {
    settings = {
      args = {
        "--extend-select=W,COM,ICN",
        "--ignore=E501,E722,COM812",
      },
    },
  },
})
```

**Reply by jacksonneal** [reply]
Curious how you are liking this after a month?  I have used pyright for some time, but have increasingly used ruff, just started with ruff-lsp and would consider replacing pyright to avoid the duplicate diagnostics if there's not a downside.

**Reply by epsentosa** [reply]
currently it is my best experience using jedi + ruff_lsp, especially on big python project, that many of files can contain more that 10 thousand lines

**Reply by benon33** [reply]
Jeez, i spent probably at least two days trying to figure out why pyright wasn't working. I am now also using jedi and ruff like you. thank you goat

**Reply by StitiFatah** [reply]
May I ask what are you using for typing errors ? I'm trying jedi for lsp and mypy for typing but well there are still duplicates between ruff and mypy, even after using your snippets. I've also installed and checked your config (and installed ruff+jedi) but didn't find anything.

**Reply by epsentosa** [reply]
<img width="743" alt="Screenshot 2024-03-24 at 15 42 41" src="https://github.com/microsoft/pyright/assets/96678035/df4a2603-a816-488c-8b52-5e643a0cf04c">

for typing error, u can use mypy via plugin null-ls (now [none-ls](https://github.com/nvimtools/none-ls.nvim))

**Reply by StitiFatah** [reply]
I found mypy to be subpar (was using nvim-lint instead of none-ls but doesn't matter) compared to pyright, typing errors that pyright caught weren't caught by mypy. 

I'm now using ruffl_lsp + pyright and just ignore sepcific duplicates through ruff's config for erros and through pyright's capabilities for hints.
